### PR TITLE
Detecting and removing unused code

### PR DESCRIPTION
**Pull request by bieniusa**
Using xref, I was identifying unused exports. Exports only used in tests are now wrapped in test macros.

The current status covers everything but the `erlang_types` subdirectory.

You can get the list by running dialyzer (or `make check`).
I will exclude from the `xref` check:
- `ety_main:main/1`
- `parse:parse_transform/2`

@skogsbaer / @albsch : Can you please check if all others can/should be deleted?

### CODE REVIEW

**Review by skogsbaer — APPROVED** [review]

**Comment by bieniusa** [comment]
I am working on it - need to fix some dialyzer issues in the tests.

**Comment by skogsbaer** [comment]
@bieniusa I merged that latest changes from head (which remove list_uniq and lists_enumerate from src/utils.erl because they are also provided by all relevant versions of OTP), but now dialyzer fails.

**Comment by bieniusa** [comment]
Why is this one failing?

test/tycheck_simple_tests.erl
Line 131 Column 1: The pattern <_Name, 'all'> can never match the type <string(),{'exclude',sets:set(_)}>
Line 132 Column 1: The pattern <Name, {'include', Set}> can never match the type <string(),{'exclude',sets:set(_)}>

**Comment by bieniusa** [comment]
I am done, but can someone check that the `test` target in the Makefile does now what it intended?
The `testtest` target was probably not correct or I misunderstood it.

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ clean:
 	$(REBAR) clean
 	rm -rf _build _etylizer rebar.lock
 
-test: build testtest
+test: build #testtest
```
Comment (skogsbaer) [inline Makefile:17]: `testtest` runs the unit tests for the tests.

Background: when writing programs that should type check (as a unit test for the type checker), the program sometimes does not behave as the author of the test intended. Hence, we write unit tests to check that the tests does what it is supposed to do.

**Review by skogsbaer — APPROVED** [review]
The `testtest` target should be enabled again.

**Review by skogsbaer — CHANGES_REQUESTED** [review]
The testtest target must come back

**Review by skogsbaer — APPROVED** [review]

**Review by albsch — COMMENTED** [review]
.

**Pull request merged**